frontend/src/services/Api.ts

```diff
@@ -328,7 +329,11 @@ export const getNamespaceAppHealth = (
   if (queryTime) {
     params.queryTime = String(queryTime);
   }
-  return newRequest<NamespaceAppHealth>(HTTP_VERBS.GET, urls.namespaceHealth(namespace), params, {}).then(response => {
+  var nsHealthUrl = urls.namespaceHealth(namespace);
+  if (cluster !== '') {
+    nsHealthUrl = nsHealthUrl + '?cluster=' + cluster;
+  }
```

Comment:
nit: this might break if other params are passed. Instead of modifying the URL you can add cluster to `params` and it'll get handled correctly.

Reply:
Sure, it's true... my bad